lab-03_mergesort/src/main.c

```diff
@@ -0,0 +1,70 @@
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "../include/mergesort.h"
+
+int comporator_int(const void* a, const void* b){
+	return *(int*)a - *(int*)b;
+}
+
+int comporator_char(const void* a, const void* b){
+	return *(char*)a - *(char*)b;
+}
+
+int comporator_str(const void* a, const void* b){
+	return strcmp(*(char**)a, *(char**)b);
+}
+
+int main(int argc, char* argv[]){
+	size_t n = argc - 2;
+	if (strcmp(argv[1], "int") == 0){
+		int a[n];
```

Comment:
1) Нейминг, никаких `a` из спротпроги, только говорящие имена
2) Такое выделение забабахает тебе весь массив на стеке. Если приложению передать слишком много параметров, стек тупо переполнится. Понятно, что сценарий маловероятный, но теоретически все-таки возможный (размер буффера для хранения агрументов приложения на линухе - около 2мб, а вот размер стека функции - всего 1 мб). Так что давай-ка лучше использовать `malloc`. Ну и не забудь про проверку, связанную с ним, сам писать ее не хочу:

```suggestion
		int* array = malloc(n * sizeof(int));
```

Reply:
а какую проверку лучше делать: array == NULL или !array?